packages/workspace/src/core/file-graph/project-file-map.ts

```diff
@@ -23,7 +23,7 @@ export function createProjectFileMap(
         if (seen.has(f.file)) {
           return;
         }
-        if (f.file.startsWith(p.root)) {
+        if (f.file.startsWith(p.root || p.sourceRoot)) {
```

Comment:
Would it be safer to use `?? p.sourceRoot` to avoid undesired matching of the root defined as an empty string `""`?

Reply:
@edbzn No, that exactly the case that needs to be handled. If p.root is `""`, then this if condition will always be truthy.

Reply:
Nevermind! I thought it was for an undefined root. :+1: